### PR TITLE
fix: handle agent errors gracefully

### PR DESCRIPTION
**Pull request by andrewrynhard**
Instead of exiting (which causes a kernel panic), we shutdown instead.

Signed-off-by: Andrew Rynhard <andrew@rynhard.io>

### CODE REVIEW

File: app/metal-controller-manager/cmd/agent/main.go [hunk 1]
```diff
@@ -178,57 +178,75 @@ func reconcileIPs(endpoint string, s *smbios.Smbios, ips []net.IP) error {
 	return nil
 }
 
+func shutdown(err error) {
+	if err != nil {
```
Comment (smira) [inline app/metal-controller-manager/cmd/agent/main.go:182]: I guess we should print the error here?

**Comment by smira** [comment]
not related to this PR, but one little change I wish we could do is to extract grpc client creation to the common place, so that every function doesn't have to do that.

**Review by smira — APPROVED** [review]

**Comment by smira** [comment]
one more idea is to wrap the `main` into a function which returns an error and do `shutdown(f())`

**Comment by andrewrynhard** [comment]
> one more idea is to wrap the `main` into a function which returns an error and do `shutdown(f())`

Sorry, I missed this. The code could use some cleanup.

**Pull request merged**